Source/PercentageGradient.swift

```diff
@@ -7,19 +7,55 @@ public enum GradientDirection {
 
 }
 
+public struct GradientPoints {
+    
+    public var start: CGPoint
+    public var end: CGPoint
+    
+}
+
+public enum GradientAngle {
+    
+    case diagonalAscending
```

Comment:
Are both `diagonalAscending` and `diagonalDescending` needed? If I'm understanding this correctly, we should be able to flip from `GradientDirection.darkToLight` or `GradientDirection.lightToDark` to achieve the same effect. Please correct me if I'm wrong though!

Reply:
Yeah, they are both needed, a you can use either with .lightToDark or .darkToLight.  They don't have anything to do with color, but a different start/end point.  Ascending means that the start point is in the bottom left and end point top right, and Descending is the opposite, starts top left ends bottom right.

Reply:
Ah, I see. Is there possibly a better name to encapsulate this concept then? Just spitballing, but maybe `bottomLeftStarting` and `topLeftStarting`. To be honest, I don't know actually think those are great names, but am not sure what I'd suggest, but ideally I'd prefer it to be more obvious without needing documentation to explain it.

Reply:
I'll see what I can come up with